src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.IntegrationAccount.tests.ps1

```diff
@@ -1264,5 +1264,666 @@ InModuleScope Arcus.Scripting.IntegrationAccount {
+
+                Mock Get-Content {
+	                return [pscustomobject] @{
+		                name = $agreementName;
+		                properties = [pscustomobject] @{
+			                agreementType = 'AS2';
+			                hostPartner = 'Partner1';
+			                guestPartner = 'Partner2';
+			                hostIdentity = [pscustomobject] @{
+				                qualifier = '1';
+				                value = '12345';
+			                }
+			                guestIdentity = [pscustomobject] @{
+				                value = '98765';
+			                }
+			                content = [pscustomobject] @{
+				                aS2 = [pscustomobject] @{}
+			                }
+		                }
+	                } | ConvertTo-Json -Depth 5;
+                } -Verifiable
+
+                Mock Get-AzIntegrationAccount {
+                    return [pscustomobject]@{ Id = $integrationAccountResourceId; Name = $integrationAccountName; Type = 'Microsoft.Logic/integrationAccounts'; Location = 'westeurope'; Sku = 'Free' }
+                } -Verifiable
+
+                # Act
+                { 
+                   Set-AzIntegrationAccountAgreements -ResourceGroupName $resourceGroupName -Name $integrationAccountName -AgreementFilePath $agreementFilePath
+                } | Should -Throw -ExpectedMessage "Cannot upload Agreement to Azure Integration Account '$integrationAccountName' because the guest identity qualifier is empty"
+
+                # Assert
+                Assert-VerifiableMock
+            }
+            It "Not Providing an Guest Identity Value should fail" {
+                # Arrange
+                $subscriptionId = [guid]::NewGuid()
+                $resourceGroupName = "rg-infrastructure"
+                $integrationAccountName = "unexisting-integration-account"
+                $integrationAccountResourceId = "/subscriptions/$subscriptionId/resourceGroups/$resourceGroupName/providers/Microsoft.Logic/integrationAccounts/$integrationAccountName"
+                $agreementName = 'Dummy_New_Agreement'
+                $agreementResourceId = "/subscriptions/$subscriptionId/resourceGroups/$resourceGroupName/providers/Microsoft.Logic/integrationAccounts/$integrationAccountName/agreements/$agreementName"
+                $agreementFilePath = "$PSScriptRoot\Files\IntegrationAccount\Agreements\$agreementName.json"
+
+                Mock Get-Content {
+	                return [pscustomobject] @{
+		                name = $agreementName;
+		                properties = [pscustomobject] @{
+			                agreementType = 'AS2';
+			                hostPartner = 'Partner1';
+			                guestPartner = 'Partner2';
+			                hostIdentity = [pscustomobject] @{
+				                qualifier = '1';
+				                value = '12345';
+			                }
+			                guestIdentity = [pscustomobject] @{
+				                qualifier = '1';
```

Comment:
Uh, this test is called to verify that no guest identity is passed, that it fails, but this stubbed object contains a guest identity? 😅

Reply:
There are 2 parts to the identity, this one checks the identity value, the other test checks the identity qualifier. Should be correct no?

Reply:
Aha! Now I see. I was looking for a test where the `guestIdentity` itself was being set to `$null` or something.